autoscheduler/frontend/src/tests/ui/ConfigureCard.test.tsx

```diff
@@ -1,19 +1,42 @@
+import fetchMock, { enableFetchMocks } from 'jest-fetch-mock';
+
+enableFetchMocks();
+
+/* eslint-disable import/first */ // enableFetchMocks must be called before others are imported
+
 import * as React from 'react';
 import { render, fireEvent } from '@testing-library/react';
-import { createStore } from 'redux';
+import { createStore, applyMiddleware } from 'redux';
 import { Provider } from 'react-redux';
+import thunk from 'redux-thunk';
 import ConfigureCard from '../../components/SchedulingPage/ConfigureCard/ConfigureCard';
-import fetch from '../../components/SchedulingPage/ConfigureCard/generateSchedulesMock';
 import autoSchedulerReducer from '../../redux/reducer';
+import { updateCourseCard } from '../../redux/actions/courseCards';
+import { CustomizationLevel } from '../../types/CourseCardOptions';
 
 jest.mock('../../components/SchedulingPage/ConfigureCard/generateSchedulesMock', () => ({
   __esModule: true,
   default: jest.fn(() => new Promise(() => {})),
 }));
 
-describe('OptionsCard component', () => {
+describe('ConfigureCard component', () => {
+  beforeEach(fetchMock.mockReset);
+
   describe('makes an API call', () => {
     test('when the user clicks Fetch Schedules', () => {
+      let fetchCount = 0;
+
+      fetchMock.mockImplementation((route: string): Promise<Response> => {
```

Comment:
If there are no other calls to `fetch` in this test, we might be able to `expect(fetchMock).toBeCalledWith('/scheduler/generate')`

Reply:
So I thought this would work too, but since it's a POST request with a JSON body, we'd also have to assert the body of it:

```
 FAIL  tests/ui/ConfigureCard.test.tsx (6.81s)
  ● ConfigureCard component › makes an API call › when the user clicks Fetch Schedules

    expect(jest.fn()).toHaveBeenCalledWith(...expected)

    Expected: "scheduler/generate"
    Received: "scheduler/generate", {"body": "{\"term\":\"\",\"includeFull\":false,\"courses\":[],\"availabilities\":[]}", "method": "POST"}

    Number of calls: 1
```

Reply:
In that case, try `expect(fetchMock).toBeCalledWith('/scheduler/generate', expect.any())`